custom_components/bhyve/sensor.py

```diff
@@ -225,3 +230,49 @@ def _on_ws_data(self, data):
 
     def _should_handle_event(self, event_name, data):
         return event_name in [EVENT_CHANGE_MODE]
+
+
+class BHyveTempSensor(BHyveDeviceEntity):
+    """Define a BHyve sensor."""
+
+    def __init__(self, hass, bhyve, device):
+        """Initialize the sensor."""
+        name = "{0} temp sensor".format(device.get("name"))
+        _LOGGER.info("Creating state sensor: %s", name)
+        super().__init__(hass, bhyve, device, name, "thermometer", DEVICE_CLASS_TEMPERATURE)
+
+    def _setup(self, device):
+        self._state = device.get("status", {}).get("temp_f")
```

Comment:
Can you please look into [Sensor Temperature Conversion](https://developers.home-assistant.io/blog/2021/08/12/sensor_temperature_conversion/) and how this entity can use this to convert between ºC and ºF automatically?

Thanks

Reply:
Oddly enough, the API only provides me with a field named “temp_f”, there is no “temp_c”, at least in my account. Maybe this is automatic based on my provided address in the Bhyve app?

Reply:
Nope - in my experience I get imperial measurements too (eg `consumption_gallons`), and I'm in Australia.

I believe all you need to change the logic of this sensor entity to implement the `native_value` and `native_unit_of_measurement` functions:

```py
def native_value(self):
    return self._state

def native_unit_of_measurement(self):
    return TEMP_FAHRENHEIT
```

(and then not implement `state()`.

Reply:
https://developers.home-assistant.io/blog/2021/08/12/sensor_temperature_conversion/

Reply:
Testing...

Reply:
I removed the state property from the entity and added these 2 but now its value is coming in as unknown.

Reply:
I think it's only available from v `2021.09` - see https://www.home-assistant.io/blog/2021/09/01/release-20219/#breaking-changes

Reply:
So as the PR stands, it’s not woking for me with the ‘native’ conversation. Maybe someone else can take a look at it? I changed mine back to ‘def unit_of_measurment’ and ‘def state’ and its working perfectly.

Reply:
Which version of HA are you running?

Reply:
2021.9.2

Reply:
Ok this is likely because this class extends `BhyveDeviceEntity` which implements `Entity` and not `SensorEntity`

Reply:
@disforw any chance you can take a look at doing this? Keen to get this merged and into the hands of the community 😁